### PR TITLE
feat: add project filtering and grouping to daily command

### LINKED ISSUE #137: Add support for multiple Claude Code instances running at the same time
I often run multiple instances of Claude Code in parallel, working on different projects. Can you add an option to display the usage data of each process separately?

### PR DESCRIPTION
**Pull request by phrazzld**
## Summary

Adds project filtering and grouping support to the daily command for tracking Claude Code usage across multiple parallel sessions.

• **Project grouping**: New `--instances` flag displays daily usage breakdown by project
• **Project filtering**: New `--project` flag filters daily results to specific projects  
• **Smart aliases**: Configure custom project names via `CCUSAGE_PROJECT_ALIASES` env var
• **Enhanced JSON output**: Project-grouped data structure for daily reports
• **Improved path parsing**: Better extraction of meaningful project names from directory structures

Implementation focuses specifically on the daily command with comprehensive project name handling.

## Implementation Details

- New `_project-names.ts` module with intelligent project name parsing and alias support
- Enhanced daily data loader with project-aware filtering and grouping
- Extended daily command interface with `--instances` and `--project` flags
- Comprehensive test coverage for project name formatting edge cases
- Backward compatible - existing daily command usage patterns unchanged

## Usage Examples

```bash
# Show daily usage by project
ccusage daily --instances

# Filter daily report to specific project 
ccusage daily --project ccusage

# Custom project aliases for daily reports
export CCUSAGE_PROJECT_ALIASES="long-project-name=Short Name"
ccusage daily --instances --json
```

Closes #137

🤖 Generated with [Claude Code](https://claude.ai/code)

## Summary by CodeRabbit

* **New Features**
  * Added the ability to group and filter usage data by project using new command-line options (`--instances`, `--project`) in daily reports.
  * Introduced support for custom project name aliases via the `CCUSAGE_PROJECT_ALIASES` environment variable for clearer reporting.
  * Enhanced JSON and table outputs to display usage breakdowns per project, with improved formatting for project names.

* **Documentation**
  * Updated user guides and CLI documentation to explain project grouping, filtering, aliasing, and new output formats.
  * Added detailed examples and usage scenarios for project-level reporting and JSON analysis.

* **Style**
  * Minor formatting and comment improvements in code for clarity.

### CODE REVIEW

**Comment by phrazzld** [comment]
without `--instances` flag:
Screenshot 2025-06-23 at 7 09 16 PM

with `--instances` flag, for sessions without good names:
Screenshot 2025-06-23 at 7 09 48 PM

with `--instances` flag, for sessions with good names, tail:
Screenshot 2025-06-23 at 7 10 29 PM

happy to continue iterating on this feature with feedback. i like being able to see the breakdown by session, but this current implementation still feels a tad unapproachable.

better session names would be nice, but i'm not sure how much i can do there. (probably something though.)

support for the `--instances` flag with other commands might also be nice. there might be better fits than `daily` (i'd certainly like to see the live breakdown), but i wanted to keep this pull request as tight and focused as i could.

**Comment by ryoppippi** [comment]
It really looks cool. I'll check it out!

File: src/_consts.ts [hunk 1]
```diff
@@ -67,6 +67,13 @@ export const DEFAULT_CLAUDE_CONFIG_PATH = `${XDG_CONFIG_DIR}/claude`;
  */
 export const CLAUDE_CONFIG_DIR_ENV = 'CLAUDE_CONFIG_DIR';
 
+/**
+ * Environment variable for custom project name aliases
+ * Format: "raw-name=alias,another-name=other-alias"
+ * Used to override display names for project directories
+ */
+export const PROJECT_ALIASES_ENV = 'CCUSAGE_PROJECT_ALIASES';
```
Comment (ryoppippi) [inline src/_consts.ts:91]: Why we define this environment variable.
Is it defined in Claude Code implementation?

Comment (moomooskycow) [reply src/_consts.ts:91]: because claude code doesn't always have a nice name for every project/session (it's sometimes just a uuid), this is a way to manually define mappings for those claude code session uuids to something more human readable

there are definitely improvements we could make here, smarter defaults we could use. but i'm not sure the best way to handle it at the moment and it would certainly add complexity, so i'm hesitant to address it directly in this pr

**Comment by ryoppippi** [comment]
@phrazzld if you are ready, let me know. mention me @ryoppippi

File: src/_consts.ts (hunk 1 above)
Comment (ryoppippi) [reply src/_consts.ts:91]: Ah got it.
Hmm it looks fine for now. I'll merge it and we gonna think about it

**Comment by ryoppippi** [comment]
@phrazzld 
Thanks
the implementation looks brilliant! nice!

Lastly, update the documantation. I'll check the docs and will merge it!
Thank you for your contribution

**Comment by phrazzld** [comment]
@ryoppippi just pushed some updates to the documentation.

let me know if there's anything else i can do!

**Comment by ryoppippi** [comment]
@phrazzld 
Hi!
Sorry there is a conflict happens.
So sorry could you work on it. I'm about to merge it!

**Comment by ryoppippi** [comment]
there is a conflict! I'll work on resolving tomorrow!
but if you can please do!
I'll release your improvements soon!!!

**Comment by phrazzld** [comment]
good call. made a few updates, let me know if there's anything else you would like from this pull request!

**Comment by ryoppippi** [comment]
thank you for your continuous work. I'll review it later today(UK time)

**Comment by ryoppippi** [comment]
if it is done, perfect

**Comment by ryoppippi** [comment]
again, thank you for your contribution!!
I think we need to fix the live monitor one

**Comment by ryoppippi** [comment]
ok so the layout issue is not yours. I'll merge it

**Review by ryoppippi — APPROVED** [review]
LGTM

**Comment by phrazzld** [comment]
hey, apologies for the late response, i was away for the week. thanks for the review and the merge!
